db.go

```diff
@@ -102,8 +102,11 @@ type DB struct {
 	statlock sync.RWMutex // Protects stats access.
 
 	ops struct {
-		writeAt func(b []byte, off int64) (n int, err error)
+		writeAt  func(b []byte, off int64) (n int, err error)
+		Truncate func(size int64) error
```

Comment:
This `ops` structure was added for mocking tests but didn't really get fleshed out enough. I'd rather see the check for `readOnly` done in each of the `mmap()` functions and then avoid `Truncate()` there if `true`. It's easier to read and there's not so much magic going on.